### PR TITLE
Feature/youngjae/upscale

### PR DESCRIPTION
**Pull request by byeongal**

### CODE REVIEW

File: src/api/inference.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    window_size = 8
+    scale = MODEL_INFO[model_settings.model_name].scale
+    img_lq = img_lq = cv2.imread(f"{task_id}.png", cv2.IMREAD_COLOR).astype(np.float32) / 255.0
+    img_lq = np.transpose(img_lq if img_lq.shape[2] == 1 else img_lq[:, :, [2, 1, 0]], (2, 0, 1))  # HCW-BGR to CHW-RGB
+    img_lq = torch.from_numpy(img_lq).float().unsqueeze(0).to(model_settings.device)  # CHW-RGB to NCHW-RGB
+    # inference
+    with torch.no_grad():
+        # pad input image to be a multiple of window_size
+        _, _, h_old, w_old = img_lq.size()
+        h_pad = (h_old // window_size + 1) * window_size - h_old
```
Comment (589hero) [inline src/api/inference.py]: model의 forward에서 padding을 넣어주는 거 같은데 직접 padding 추가는 안 해도 되지 않을까요?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ENV PYTHONUNBUFFERED=1 \
+    PYTHONDONTWRITEBYTECODE=1 \
+    POETRY_HOME="/opt/poetry" \
+    POETRY_VIRTUALENVS_IN_PROJECT=true
+
+ENV PATH="$POETRY_HOME/bin:$VIRTUAL_ENVIRONMENT_PATH/bin:$PATH"
+
+# Install Poetry
+# https://python-poetry.org/docs/#osx--linux--bashonwindows-install-instructions
+RUN apt-get update \
```
Comment (589hero) [inline Dockerfile]: 위에서 실행한 커맨드들은 입력 안 해도 되지 않을까요?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    && python3.8 get-pip.py
+
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.8 1 \
+    && update-alternatives --install /usr/local/bin/pip pip /usr/local/bin/pip3.8 1
+
+RUN python3.8 -m venv /home/venv
+
+ENV PATH="/home/venv/bin:$PATH"
+
+RUN python -m pip install -U pip setuptools
```
Comment (589hero) [inline Dockerfile:28]: `poetry install`하면 가상환경이 같이 생성되어서 이 부분은 삭제하는게 어떤가요?
- Reference : https://github.com/ainize-team/LLM-FastAPI/blob/develop/Dockerfile

File: src/api/inference.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        img_lq = torch.cat([img_lq, torch.flip(img_lq, [3])], 3)[:, :, :, : w_old + w_pad]
+        output = model(img_lq)
+        output = output[..., : h_old * scale, : w_old * scale]
+
+    # save image
+    output = output.data.squeeze().float().cpu().clamp_(0, 1).numpy()
+    if output.ndim == 3:
+        output = np.transpose(output[[2, 1, 0], :, :], (1, 2, 0))  # CHW-RGB to HCW-BGR
+    output = (output * 255.0).round().astype(np.uint8)  # float32 to uint8
+    cv2.imwrite(f"{task_id}_SwinIR.png", output)
```
Comment (589hero) [inline src/api/inference.py:63]: minor) `f"{task_id}_{model_settings.model_name}.png"`처럼 모델명에 따라 이미지 이름을 다르게 하는 건 어떤가요?

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    curl \
+    && rm -rf /var/lib/apt/lists/* \
+    && cd /tmp \
+    && curl -O https://bootstrap.pypa.io/get-pip.py \
+    && python3.8 get-pip.py
+
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.8 1 \
+    && update-alternatives --install /usr/local/bin/pip pip /usr/local/bin/pip3.8 1
+
+RUN python3.8 -m venv /home/venv
```
Comment (589hero) [inline Dockerfile]: poetry install하면 가상환경이 같이 생성되어서 이 부분은 삭제하는게 어떤가요?

Reference : https://github.com/ainize-team/LLM-FastAPI/blob/develop/Dockerfile

File: src/api/inference.py (hunk 1 above)
Comment (byeongal) [reply src/api/inference.py]: 원저자 코드에서 이렇게 해서 확인해보겠습니다.
https://github.com/JingyunLiang/SwinIR/blob/main/main_test_swinir.py#L70

File: src/api/inference.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    if file.content_type == "image/png":
+        with open(f"{task_id}.png", "wb") as f:
+            contents = await file.read()
+            f.write(contents)
+    else:
+        raise HTTPException(status_code=400, detail="Only Support PNG file")
+    model: SwinIR = request.app.state.model
+    window_size = 8
+    scale = MODEL_INFO[model_settings.model_name].scale
+    img_lq = img_lq = cv2.imread(f"{task_id}.png", cv2.IMREAD_COLOR).astype(np.float32) / 255.0
```
Comment (589hero) [inline src/api/inference.py]: minor) 타이핑 실수 같습니다

File: src/api/inference.py (hunk 4 above)
Comment (byeongal) [reply src/api/inference.py:63]: 요고 파일을 계속 남겨두면 터질거 같아서 제거하는 방향으로 가겠습니다.

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    curl \
+    && curl -sSL https://install.python-poetry.org | python3 -
+
+WORKDIR /app
+COPY ./pyproject.toml ./pyproject.toml
+COPY ./poetry.lock ./poetry.lock
+RUN poetry install --only main
+ENV PATH="/app/.venv/bin:$PATH"
+
+EXPOSE 8000
```
Comment (589hero) [inline Dockerfile:55]: 프로젝트 파일들을 복사하는 부분이 빠져 `COPY ./src/ /app/`를 추가해야될 것 같습니다.

File: src/api/inference.py (hunk 4 above)
Comment (589hero) [reply src/api/inference.py:63]: 넵넵

**Review by 589hero — APPROVED** [review]
LGTM
